### PR TITLE
[#1657] add options to hide categories and search from anonymous users

### PR DESCRIPTION
**Pull request by pi-sigma**
- added model fields (bool) for hiding categories, search bar + search page
- added `LoginMaybeRequiredMixin` to categories and search views in order to conditionally restrict access to authenticated users

Taiga: [#1657](https://taiga.maykinmedia.nl/project/open-inwoner/task/1657)

### CODE REVIEW

**Review by alextreme — APPROVED** [review]
Looks good, however considering the refactor is a bit bigger than only the requested change @jiromaykin should also take a look (and he's back tomorrow)

**Comment by jiromaykin** [comment]
> ....considering the refactor is a bit bigger than only the requested change @jiromaykin should also take a look...

I will check it out and will need to update everything locally and see what I can do here.

File: src/open_inwoner/configurations/migrations/0048_auto_20230816_1255.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="siteconfiguration",
+            name="hide_search_from_anonymous_users",
+            field=models.BooleanField(
+                default=False,
+                help_text="If checked, only authenticated users will be able to search the page.",
+                verbose_name="Hide search from anonymouns users",
```
Comment (jiromaykin) [inline src/open_inwoner/configurations/migrations/0048_auto_20230816_1255.py:22]: 1st: Spelling mistake: "anonymouns" should be "anonymous"

File: src/open_inwoner/configurations/models.py [hunk 2]
```diff
@@ -445,6 +445,20 @@ class SiteConfiguration(SingletonModel):
         default=True,
         help_text=_("Whether file sharing via the messages is allowed or not"),
     )
+    hide_categories_from_anonymous_users = models.BooleanField(
+        verbose_name=_("Hide categories from anonymouns users"),
```
Comment (jiromaykin) [inline src/open_inwoner/configurations/models.py:449]: 2nd occurence: Spelling mistake: "anonymouns" should be "anonymous"

File: src/open_inwoner/configurations/models.py [hunk 3]
```diff
@@ -445,6 +445,20 @@ class SiteConfiguration(SingletonModel):
...
+        default=False,
+        help_text=_(
+            "If checked, categories will be hidden from users who are not logged in."
+        ),
+    )
+    hide_search_from_anonymous_users = models.BooleanField(
+        verbose_name=_("Hide search from anonymouns users"),
```
Comment (jiromaykin) [inline src/open_inwoner/configurations/models.py:456]: 3rd: Spelling mistake: "anonymouns" should be "anonymous"

**Review by jiromaykin — CHANGES_REQUESTED** [review]
All else is looking good, apart from the spellign mistakes; so when logged all have become invisible, both on mobile as well as on desktop.

**Review by jiromaykin — APPROVED** [review]

**Pull request merged**